### PR TITLE
CR-1149236 Enabling passing /var/log/messages through VMR back to host

### PR DESCRIPTION
**Pull request by jeffli-xilinx**
#### Problem solved by the commit
Allow kernel logs to be dumped to /var/log/messages on APU and passed back to host through VMR

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Enhancement [CR-1149236](https://jira.xilinx.com/browse/CR-1149236)

#### Risks (if any) associated the changes in the commit
None

#### What has been tested and how, request additional testing if necessary
Tested on V70

### CODE REVIEW

File: src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c [hunk 1]
```diff
@@ -389,7 +414,7 @@ static void zchan_cmd_handler(struct platform_device *pdev, struct xgq_cmd_sq_hd
 	cmd_handler func = opcode2handler(op);
 	struct xgq_com_queue_entry r = {};
 
-	zchan_info(chan, "%s received", opcode2name(op));
+	//	zchan_info(chan, "%s received", opcode2name(op));
```
Comment (xdavidz) [inline src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c:417]: if no needed, please remove this log or switch to debug.

File: src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c [hunk 2]
```diff
@@ -216,8 +241,8 @@ static void zchan_cmd_log_page(struct zocl_rpu_channel *chan, struct xgq_cmd_sq_
 	u32 total_count = 0;
 	int ret = 0;
 
-	zchan_info(chan, "addr_off 0x%x, size %d, offset %d, pid %d",
-		add_off, size, offset, pid);
+	//	zchan_info(chan, "addr_off 0x%x, size %d, offset %d, pid %d",
```
Comment (xdavidz) [inline src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c:244]: if no needed, please remove this log or switch to debug.

File: src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c [hunk 3]
```diff
@@ -183,22 +185,45 @@ static int zchan_cmd_log_apu_log(struct zocl_rpu_channel *chan, u32 add_off, u32
...
+	fp_size = i_size_read(file_inode(fp));
+	fp_offset = offset;
+	remain_size = fp_size - offset;
+	// Send chunks of log
+	if (remain_size >= size) {
+		ret = kernel_read(fp, &log_buf, size, &fp_offset);
+		if (!ret) {
+			pr_err("Can't read from /var/log/messages\n");
+			*total_countp = 0;
+			return -EINVAL;
```
Comment (xdavidz) [inline src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c:210]: I think we need to close the fp before return.

File: src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c [hunk 4]
```diff
@@ -183,22 +185,45 @@ static int zchan_cmd_log_apu_log(struct zocl_rpu_channel *chan, u32 add_off, u32
...
+		}
+		count = size;
 	} else {
-		count = 0;
+		if(remain_size > 0) {
+			ret = kernel_read(fp, &log_buf, remain_size, &fp_offset);
+			if (!ret) {
+				pr_err("Can't read from /var/log/messages\n");
+				*total_countp = 0;
+				return -EINVAL;
```
Comment (xdavidz) [inline src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c:219]: filp_close before return?

**Review by xdavidz — APPROVED** [review]

**Comment by dayeh-xilinx** [comment]
retest this please.

File: src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c (hunk 1 above)
Comment (jeffli-xilinx) [reply src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c]: Switched to zchan_dbg

File: src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c (hunk 3 above)
Comment (jeffli-xilinx) [reply src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c:211]: updated

File: src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c (hunk 4 above)
Comment (jeffli-xilinx) [reply src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c:221]: updated

**Review by chvamshi-xilinx — APPROVED** [review]

**Pull request merged**